addons/controllers/antrea/antreaconfig_util.go

```diff
@@ -0,0 +1,143 @@
+
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cni/v1alpha1"
+)
+
+// infrastructure provider name constants
+const (
+	InfrastructureProviderVSphere = "vsphere"
+	InfrastructureProviderAWS     = "aws"
+	InfrastructureProviderAzure   = "azure"
+)
+
+const (
+	// InfrastructureRefVSphere is the vSphere infrastructure
+	InfrastructureRefVSphere = "VSphereCluster"
+	// InfrastructureRefAWS is the AWS infrastructure
+	InfrastructureRefAWS = "AWSCluster"
+	// InfrastructureRefAzure is the Azure infrastructure
+	InfrastructureRefAzure = "AzureCluster"
+)
+
+// AntreaConfigSpec defines the desired state of AntreaConfig
+type antreaConfigSpec struct {
+	InfraProvider string `yaml:"infraProvider"`
+
+	Antrea antrea `yaml:"antrea,omitempty"`
+}
+
+type antrea struct {
+	AntConfig antConfig `yaml:"config,omitempty"`
+}
+
+type antConfig struct {
+	ServiceCIDR string `yaml:"serviceCIDR,omitempty"`
+
+	ServiceCIDRv6 string `yaml:"serviceCIDRv6,omitempty"`
+
+	TrafficEncapMode string `yaml:"trafficEncapMode,omitempty"`
+
+	NoSNAT bool `yaml:"noSNAT,omitempty"`
+
+	DisableUdpTunnelOffload bool `yaml:"disableUdpTunnelOffload,omitempty"`
+
+	DefaultMTU string `yaml:"defaultMTU,omitempty"`
+
+	TlsCipherSuites string `yaml:"tlsCipherSuites,omitempty"`
+
+	FeatureGates antreaFeatureGates `yaml:"featureGates,omitempty"`
+}
+
+type antreaFeatureGates struct {
+	AntreaProxy bool `yaml:"AntreaProxy,omitempty"`
+
+	EndpointSlice bool `yaml:"EndpointSlice,omitempty"`
+
+	AntreaPolicy bool `yaml:"AntreaPolicy,omitempty"`
+
+	FlowExporter bool `yaml:"FlowExporter,omitempty"`
+
+	Egress bool `yaml:"Egress,omitempty"`
+
+	NodePortLocal bool `yaml:"NodePortLocal,omitempty"`
+
+	AntreaTraceflow bool `yaml:"AntreaTraceflow,omitempty"`
+
+	NetworkPolicyStats bool `yaml:"NetworkPolicyStats,omitempty"`
+}
+
+func getServiceCIDR(cluster *clusterapiv1beta1.Cluster) (string, error) {
+	var serviceCIDR string
+	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.Services != nil && len(cluster.Spec.ClusterNetwork.Services.CIDRBlocks) > 0 {
+		serviceCIDR = cluster.Spec.ClusterNetwork.Services.CIDRBlocks[0]
+	} else {
+		return "", errors.New("Unable to get cluster serviceCIDR")
+	}
+
+	return serviceCIDR, nil
+}
+
+func getInfraProvider(cluster *clusterapiv1beta1.Cluster) (string, error) {
+	var infraProvider string
+
+	infraProvider = cluster.Spec.InfrastructureRef.Kind
+
+	switch infraProvider {
+	case InfrastructureRefVSphere:
+		infraProvider = InfrastructureProviderVSphere
+	case InfrastructureRefAWS:
+		infraProvider = InfrastructureProviderAWS
+	case InfrastructureRefAzure:
+		infraProvider = InfrastructureProviderAzure
+	default:
+		infraProvider = InfrastructureProviderVSphere
+	}
+
+	return infraProvider, nil
+}
+
+func mapAntreaConfigSpec(cluster *clusterapiv1beta1.Cluster, config *cniv1alpha1.AntreaConfig) (*antreaConfigSpec, error) {
+	configSpec := &antreaConfigSpec{}
+
```

Comment:
@vijaykatam , @danniel1205 , I have added the logic for infra provider and service CIDR.
Please review and let me know if it the right way to derive infra provider.

I have introduced this util file to add the types similar to ( [reference](https://github.com/vmware-tanzu/tanzu-framework/pull/1553/files#diff-502f153f17c6f3a7c656dac5c8904f0397bb02f35731bc2a39f47ad98375cb8f) 

Still working on deriving dual stack for serviceCIDR and serviceCIDRv6

cc: @maralavi

Reply:
Suggestion: Ideally ```getInfraProvider()``` and ```getServiceCIDR()``` should be moved up to a util to be usable by other addons too.
Suggestion: ```InfrastructureRefDocker``` should be added too.

Reply:
@maralavi , let me move it up to a util so that it can be shared.
Yes, need to add InfrastructureRefDocker . I was working on tests and realized nilness check should be added. 
Thanks for suggestions. will add.